src/util_misc.cpp

```diff
@@ -470,6 +477,15 @@ bool Util::isValidFd(int fd)
   return _real_fcntl(fd, F_GETFL, 0) != -1;
 }
 
+// On failure, returns 0, with errno == EFAULT
+bool Util::isReadableMemoryAddress(const char *addr) {
+  // FIXME: We wanted to call libc:open(), but _real_open goes to second wrapper
+  int fd = _real_syscall(SYS_open, "/dev/null", O_WRONLY);
+  int rc = write(fd, addr, 1);
```

Comment:
I'm copying the history for this line.  It was lost in the new revision to the code.  @karya0 had suggested a shorter version of the code, that avoids the call to open():
`int rc = write(PROTECTED_STDERR_FD, addr, 0)`

@karya0 wrote:
Doing a write of "0" bytes is not dangerous. It's a well known trick and is used quite generously. The POSIX standard documents this behavior too. I would strongly suggest to use this instead of doing an open and close.

@gc00 replied:
I looked it up, and I'm getting a different reading of the POSIX standard. I'm seeing:
Implementations are allowed, but not required, to perform error checking for write() requests of zero bytes.
at: http://pubs.opengroup.org/onlinepubs/009695399/functions/write.html
This seems to say that a write of zero bytes from a bad memory address can succeed.

@gc00 also added the statement `close(fd);', (below) which he had forgotten in the earlier revision.